packages/streaming-image-volume-loader/src/helpers/sortImageIdsAndGetSpacing.ts

```diff
@@ -126,11 +126,22 @@ export default function sortImageIdsAndGetSpacing(
       Math.floor(imageIds.length / 2);
   }
 
-  const { imagePositionPatient: origin } = metaData.get(
+  const { imagePositionPatient: origin, sliceThickness } = metaData.get(
     'imagePlaneModule',
     sortedImageIds[0]
   );
 
+  // if zspacing = 0, use sliceThickness if possible
+  if (zSpacing === 0) {
+    if (sliceThickness) {
+      console.log('Could not calculate zSpacing. Using sliceThickness');
+      zSpacing = sliceThickness;
+    } else {
+      console.log(
+        'Could not calculate zSpacing. The VolumeViewport is compromised.'
+      );
+    }
+  }
```

Comment:
This should only be for the `NM`, put a comment above it please and also explain the weirdness for NM

Reply:
I dont agree. We dont know for sure that this only happens in NM

Reply:
@wayfarer3130 what do you think?

Reply:
The NM behaviour is quite complex, and in fact you can't assume that the slice thickness even corresponds to positioning of elements.  There is a whole module in DICOM dedicated to how the slices are ordered/managed - it could be multiple volumes, with several at the same position, differing in time.
I don't know enough offhand to understand what the spacing is for Tomo MG, but those also have different setup for positioning.